Fix: observation space Dict for non-concatenated groups only keeps last term

CSCSX opened the pull request:
# Description

This PR fixes a bug in the observation space construction for non-concatenated groups in `ManagerBasedRLEnv._configure_gym_env_spaces` method. Previously, only the last term in each group was included in the Dict, causing loss of observation information. Now, all terms are correctly added to the group Dict.

Fixes #3133 

<!-- As a practice, it is recommended to open an issue to have discussions on the proposed pull request.
This makes it easier for the community to keep track of what is being developed or added, and if a given feature
is demanded by more than one party. -->

## Type of change

<!-- As you go through the list, delete the ones that are not applicable. -->

- Bug fix (non-breaking change which fixes an issue)


<!--
Example:

| Before | After |
| ------ | ----- |
| _gif/png before_ | _gif/png after_ |

To upload images to a PR -- simply drag and drop an image while in edit mode and it should upload the image directly. You can then paste that source into the above before/after sections.
-->

ooctipus commented:
Hi @CSCSX  Thank you for the fix, we definitely need to fix this bug, are you open to add a test case at `source/isaaclab/test/envs/test_manager_based_rl_env_obs_spaces.py`? : ))

CSCSX commented:
Thanks for the feedback! I've added a test case in `source/isaaclab/test/envs/test_manager_based_rl_env_obs_spaces.py` that verifies all observation terms are correctly included in non-concatenated groups (not just the last one). The test uses the Franka stack environment which has multiple non-concatenated observation groups. Updated the commit with `--amend`.

ooctipus commented:
@CSCSX Congratulation on you first PR!
Happy to review your PR in future again!
